### PR TITLE
Add anomaly detection example doc

### PR DESCRIPTION
**Pull request by omeroguz45**
Added the doc for the anomaly detection example with the file anomaly-detection.md and a couple of images for the doc.

### CODE REVIEW

File: docs/anomaly-detection.md [hunk 1]
```diff
@@ -1 +1,285 @@
...
+    
+    # We calculate the average and return the ideal eps value.
+    eps_scores = eps_scores / size_0
+
+    ideal_eps = range_eps[np.where(eps_scores.max)[0][0]]
+
+    return ideal_eps
+```
+
+This is what is needed to find anomalies on a given model. Now we need to set up two more functions to get a Slack bot to message us some information and a plot of the anomalies in our model.
```
Comment (mkam039) [inline docs/anomaly-detection.md:186]: This section has a lot of information. So we need to make a summary of steps that were taken.

File: docs/anomaly-detection.md [hunk 2]
```diff
@@ -1 +1,285 @@
...
+
+## Sending results via Slack
+The two functions we need are `plot_anomalies` and `send_slack_file`, which is present in [metric forecast](metric-forecast.d) and [slack bot](slack-example.md) examples.
```
Comment (mkam039) [inline docs/anomaly-detection.md:189]: How is this plot function related to the metric forecast function? Do you want to say something like "analogous to ..."?

File: docs/anomaly-detection.md [hunk 3]
```diff
@@ -1 +1,285 @@
...
+            file=file_path,
+            title="fal.ai anomaly detection",
+            initial_comment=message_text,
+        )
+    except SlackApiError as e:
+        assert e.response["error"]
+```
+
+## Finding anomalies on a dbt model with fal
+At last, we have all the functions needed for anomaly detection. However, there is a very important thing missing: Our model. Using fal we can load our model in the form of a DataFrame with ease using the extrmely handy `ref()` function and the `context` object.
```
Comment (mkam039) [inline docs/anomaly-detection.md:227]: Fix spelling

**Review by mkam039 — CHANGES_REQUESTED** [review]
Great job! Extensive example with a lot of information.

I'm not sure if "Anomaly detection" is the right title here. How about "Trend change detection" or "Wave detection".

Overall, I think that we can tighten the language and focus on simple clarity.

File: docs/anomaly-detection.md (hunk 2 above)
Comment (omeroguz45) [reply docs/anomaly-detection.md:189]: I meant the slack function, will clarify

**Comment by omeroguz45** [comment]
The file has been modified according to Meder's comments and has been [committed](https://github.com/fal-ai/fal/pull/46/commits/a3ea92fe5bea66e42436787821f37eafb900b144).

File: docs/anomaly-detection.md [hunk 4]
```diff
@@ -1 +1,290 @@
...
+
+![Slack bot message](anomaly_detection_slack_message.png)
+
+And, that's it. We have our Slack bot notifying us about the anomalous data points. Now, let's take a look at the anomalies that have been spotted on our data and do a little analysis:
+
+![Plot of anomalies](2021-11-20-2322-CET.jpg)
+
+As we can see, the system finds 3 anomalies in the first wave and 1 while the wave flattens. This behaviour is unexpected, however it is insignificant enough that only the extreme outliers are reported as anomalous. In this second wave, we can observe a much, much more sharp upwards trend that the system classifies most of the wave as anomalous.
+
+In this example, we can see the power of fal, as we can have an extremely vital and time sensitive system directly from your dbt project.
```
Comment (mkam039) [inline docs/anomaly-detection.md:290]: Let's add a sentence in the end with next steps a user could take.

**Review by mkam039 — CHANGES_REQUESTED** [review]
It's very good, Omer! Thanks for the edits.

I see a couple more small changes that could be done and then I think we're good to go.

**Comment by omeroguz45** [comment]
The file has been modified according to Meder's comments and has been [committed](https://github.com/fal-ai/fal/commit/944c1d21256aa5b1816952cffa8cc6df3e213548).

**Review by mkam039 — APPROVED** [review]
Looks good! Maybe we can update the blog post with some of these changes.
